OPHD/States/CrimeRateUpdate.cpp

```diff
@@ -27,6 +38,13 @@ void CrimeRateUpdate::update(const std::vector<TileList>& policeOverlays)
 		int crimeRateChange = isProtectedByPolice(policeOverlays, structure) ? -1 : 1;
 		structure->increaseCrimeRate(crimeRateChange);
 
+		// Crime Rate of 0% means no crime
+		// Crime Rate of 100% means crime occurs 10% of the time
+		if (structure->crimeRate() + random() > 1000)
+		{
+			mStructuresCommittingCrimes.push_back(structure);
+		}
+
```

Comment:
I hadn't thought of doing it this way... but generally I like it.

I recently added a crime icon with the thinking that structures that have high crime rates (or need maintenance) could have the icon displayed over top of them. In this case it would be displayed where crime is actually happening versus just the crime rate itself being high. I like it, simplifies that part of it a bit and boils it down to individual criminal acts that can also push notifications (you show an example of that which I really like).

Nice work!

Reply:
Glad the module separate is appreciated. Thank you.

I would encourage highlighting high crime rate as you were planning. I sort of like the idea of also marking structures that recently had major criminal activity if it wasn't too busy. I think highlighting high crime structures might be a bit higher priority since the message log will tell the player when specific criminal activity occurs.